[SOLVED] Won't reconnect after disconnect

I've been running Magic Mirror on a Raspberry Pi 3B+ for about a year now.  I've just added MMM-HomeAssistantDisplay to display energy values from Home Assistant.  After a random time, the display values stop updating.  I enabled debuglogging in the config.js for this module but didn't see any additional info.  Here's a typical mm-out.log:
```
[26.05.2022 14:18.04.433] [LOG]   Connecting socket for: MMM-HomeAssistantDisplay
[26.05.2022 14:18.04.436] [LOG]   Sockets connected & modules started ...
[26.05.2022 14:18.04.875] [LOG]   Launching application.
[26.05.2022 14:18.15.275] [LOG]   Create new calendarfetcher for url: http://www.calendarlabs.com/ical-calendar/ics/76/US_Holidays.ics - Interval: 300000
.
.
.
[26.05.2022 14:18.16.607] [INFO]  [MMM-HomeAssistantDisplay] [ 'HomeAssistant connected for module_11_MMM-HomeAssistantDisplay' ]
[26.05.2022 14:18.16.659] [INFO]  Checking git for module: MMM-Remote-Control
[26.05.2022 14:18.16.749] [LOG]   Electron's cache successfully cleared.
[26.05.2022 14:18.16.861] [INFO]  Checking git for module: WallberryTheme
[26.05.2022 14:18.16.983] [INFO]  Checking git for module: MMM-HomeAssistantDisplay
[26.05.2022 14:18.17.085] [INFO]  Checking git for module: default
[26.05.2022 14:18.17.403] [INFO]  [MMM-HomeAssistantDisplay] [
  'Conected to Home Assistant for module_11_MMM-HomeAssistantDisplay after 0 retries'
]
[26.05.2022 14:18.17.706] [INFO]  Calendar-Fetcher: Broadcasting 3 events.
[26.05.2022 14:18.18.592] [INFO]  Calendar-Fetcher: Broadcasting 5 events.
[26.05.2022 14:18.21.564] [INFO]  Calendar-Fetcher: Broadcasting 3 events.
[26.05.2022 14:18.25.384] [DEBUG] [MMM-HomeAssistantDisplay] [
  'Found listening connection (module_11_MMM-HomeAssistantDisplay) for entity sensor.envoy_202151121117_current_power_production'
]
[26.05.2022 14:18.25.451] [DEBUG] [MMM-HomeAssistantDisplay] [
  'Found listening connection (module_11_MMM-HomeAssistantDisplay) for entity sensor.envoy_202151121117_current_power_consumption'
]
[26.05.2022 14:18.25.533] [DEBUG] [MMM-HomeAssistantDisplay] [
  'Found listening connection (module_11_MMM-HomeAssistantDisplay) for entity sensor.grid_export_power'
]
[26.05.2022 14:19.25.330] [DEBUG] [MMM-HomeAssistantDisplay] [
  'Found listening connection (module_11_MMM-HomeAssistantDisplay) for entity sensor.envoy_202151121117_current_power_production'
]
[26.05.2022 14:19.25.421] [DEBUG] [MMM-HomeAssistantDisplay] [
  'Found listening connection (module_11_MMM-HomeAssistantDisplay) for entity sensor.envoy_202151121117_current_power_consumption'
]
[26.05.2022 14:19.25.436] [DEBUG] [MMM-HomeAssistantDisplay] [
  'Found listening connection (module_11_MMM-HomeAssistantDisplay) for entity sensor.grid_export_power'
]
[26.05.2022 14:20.25.528] [DEBUG] [MMM-HomeAssistantDisplay] [
  'Found listening connection (module_11_MMM-HomeAssistantDisplay) for entity sensor.envoy_202151121117_current_power_production'
]
[26.05.2022 14:20.25.608] [DEBUG] [MMM-HomeAssistantDisplay] [
  'Found listening connection (module_11_MMM-HomeAssistantDisplay) for entity sensor.envoy_202151121117_current_power_consumption'
]
[26.05.2022 14:20.25.654] [DEBUG] [MMM-HomeAssistantDisplay] [
  'Found listening connection (module_11_MMM-HomeAssistantDisplay) for entity sensor.grid_export_power'
]
[26.05.2022 14:21.25.434] [DEBUG] [MMM-HomeAssistantDisplay] [
  'Found listening connection (module_11_MMM-HomeAssistantDisplay) for entity sensor.envoy_202151121117_current_power_production'
]
[26.05.2022 14:21.25.507] [DEBUG] [MMM-HomeAssistantDisplay] [
  'Found listening connection (module_11_MMM-HomeAssistantDisplay) for entity sensor.envoy_202151121117_current_power_consumption'
]
[26.05.2022 14:21.25.601] [DEBUG] [MMM-HomeAssistantDisplay] [
  'Found listening connection (module_11_MMM-HomeAssistantDisplay) for entity sensor.grid_export_power'
]
[26.05.2022 14:21.55.554] [DEBUG] [MMM-HomeAssistantDisplay] [ 'Hass WS Disconnected (module_11_MMM-HomeAssistantDisplay)' ]
[26.05.2022 14:21.55.571] [INFO]  [MMM-HomeAssistantDisplay] [
  'Reconnecting to Home Assistant websocket for module_11_MMM-HomeAssistantDisplay'
]
[26.05.2022 14:21.55.636] [INFO]  [MMM-HomeAssistantDisplay] [
  'Unable to connect to Home Assistant for module_11_MMM-HomeAssistantDisplay: socket hang up'
]
[26.05.2022 14:23.18.397] [INFO]  Calendar-Fetcher: Broadcasting 3 events.
[26.05.2022 14:23.19.477] [INFO]  Calendar-Fetcher: Broadcasting 5 events.
[26.05.2022 14:23.25.133] [INFO]  Calendar-Fetcher: Broadcasting 3 events.
```
and mm-error.out:
```
[16288:0526/141806.801431:ERROR:gpu_process_host.cc(979)] GPU process exited unexpectedly: exit_code=139
[16288:0526/141808.357494:ERROR:gpu_process_host.cc(979)] GPU process exited unexpectedly: exit_code=139
[16288:0526/141810.036174:ERROR:gpu_process_host.cc(979)] GPU process exited unexpectedly: exit_code=139
[16406:0526/141811.693763:ERROR:sandbox_linux.cc(377)] InitializeSandbox() called with multiple threads in process gpu-process.
[26.05.2022 14:21.55.628] [ERROR] [MMM-HomeAssistantDisplay] [
  'Unable to connect to Home Assistant for module module_11_MMM-HomeAssistantDisplay failed with message: ',
  'socket hang up'
]
```
This is the module configuration from config.js with some sensitive items replaced:
```
    {
      disabled: false,
      module: "MMM-HomeAssistantDisplay",
      position: "bottom_right",
      config: {
        host: HOST,
        port: "443",
        useTLS: true,
        ignoreCert: true,
        token: TOKEN,
        title: "Energy Distribution",
        class: "energy-distribution",
        debuglogging: false,
        sections: [{
          triggerEntities: [
            "sensor.grid_import_power",
            "sensor.grid_export_power",
            "sensor.envoy_202151121117_current_power_consumption",
            "sensor.envoy_202151121117_current_power_production",
          ],
          displayTemplate: `
            <table class="medium" style="padding-top:10px;padding-right:20px;">
              <tr>
              <td align="left">Current Power Consumption&nbsp;&nbsp;</td>
              <td align="right">{{states('sensor.envoy_202151121117_current_power_consumption')}} {{state_attr('sensor.envoy_202151121117_current_power_consumption','unit_of_measurement')}}</td>
              </tr>
              <tr>
              <td align="left">Current Power Production&nbsp;&nbsp;</td>
              <td align="right">{{states('sensor.envoy_202151121117_current_power_production')}} {{state_attr('sensor.envoy_202151121117_current_power_consumption','unit_of_measurement')}}</td>
              </tr>
              <tr>
              <td align="left">{{state_attr('sensor.grid_import_power','friendly_name')}}</td>
              <td align="right">{{states('sensor.grid_import_power')}} {{state_attr('sensor.grid_import_power','unit_of_measurement')}}</td>
              </tr>
              <tr>
              <td align="left">{{state_attr('sensor.grid_export_power','friendly_name')}}</td>
              <td align="right">{{states('sensor.grid_export_power')}} {{state_attr('sensor.grid_export_power','unit_of_measurement')}}</td>
              </tr>
            </table>
            `,
          class: "energy-items"
         }]
      },
    },
```
I've pulled the latest Magic Mirror version from the repository but no difference.  Both the RPi and Home Assistant server are on the same LAN.  The HA server is proxied by HAProxy (no relation) on my firewall/router.  I'm not seeing any errors in the HA logs regarding Magic Mirror.  Let me know what additional information I can provide.

--- a/node_helper.js
+++ b/node_helper.js
@@ -142,7 +142,7 @@ async function backoffWSConnection(identifier, connectionConfig) {
 	});
 
 	call.retryIf(function(err) {
-		 return err.error.code == 'ECONNREFUSED'; 
+		 return true; 
 	});
 
 	call.setStrategy(new backoff.ExponentialStrategy({
@@ -180,6 +180,7 @@ async function reconnectWebsocket(payload) {
 		host: payload.host,
 		port: payload.port,
 		token: payload.token,
+		useTLS: payload.useTLS,
 		ignoreCert: payload.ignoreCert
 	};
 	for (const connection in this.connections) {

